### PR TITLE
Events journaling and update daemon

### PR DESCRIPTION
**Pull request by xbrianh**
This daemon is called on a schedule every five minutes. It performs as much work as it can in five minutes and then quits. Uncompleted work will be resumed on the next call.

### CODE REVIEW

File: daemons/dss-events-scribe/app.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+@app.scheduled_function("rate(5 minutes)")
+def flashflood_journal_and_update(event, context):
+    # TODO: Make this configurable
+    minimum_number_of_events = 10 if "dev" == os.environ['DSS_DEPLOYMENT_STAGE'] else 1000
+
+    def lambda_seconds_remaining():
+        # lambda time to live is configured with `lambda_timeout` in `daemons/dss-events-scribe/.chalice/config.json`
+        return context.get_remaining_time_in_millis() / 1000
```
Comment (DailyDreaming) [inline daemons/dss-events-scribe/app.py]: Nit: I think this returns a float, and I'd expect an `int` from the function name?  Would it be worth adding typing to this file?

**Comment by DailyDreaming** [comment]
Looks like just a linting error:
```
tests/test_events.py:210:9: E306 expected 1 blank line before a nested definition, found 0
tests/test_events.py:215:9: E306 expected 1 blank line before a nested definition, found 0
tests/test_events.py:225:1: W293 blank line contains whitespace
Makefile:7: recipe for target 'lint' failed
```

File: daemons/dss-events-scribe/app.py (hunk 1 above)
Comment (xbrianh) [reply daemons/dss-events-scribe/app.py]: typing added

File: iam/policy-templates/dss-events-scribe-lambda.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "arn:aws:logs:*:*:*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": "s3:*",
+      "Resource": [
+        "arn:aws:s3:::*",
```
[the remark this answers is not in the document]
Comment (xbrianh) [reply iam/policy-templates/dss-events-scribe-lambda.json]: This should be handed in a more comprehensive PR addressing existing daemons.

File: daemons/dss-events-scribe/app.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+"""
+See Readme.md in this directory for documentation on the dss-notify-v2 daemon.
```
[the remark this answers is not in the document]
Comment (xbrianh) [reply daemons/dss-events-scribe/app.py]: This was a copy-paste error.

File: daemons/dss-events-scribe/app.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                    replica.finished_journaling = True
+        else:
+            break
+
+    for replica in cycle(replicas):
+        if lambda_seconds_remaining() > 30 and not all(replica.finished_updating for replica in replicas):
+            if not replica.finished_updating:
+                number_of_updates_applied = update_flashflood(replica.prefix, minimum_number_of_events)
+                if 0 == number_of_updates_applied:
+                    replica.finished_updating = True
```
[the remark this answers is not in the document]
Comment (xbrianh) [reply daemons/dss-events-scribe/app.py]: I'll add logging.

File: daemons/dss-events-scribe/.chalice/config.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  },
+  "stages": {
+    "dev": {
+      "api_gateway_stage": "dev",
+      "tags": {},
+      "environment_variables": {}
+    }
+  },
+  "lambda_timeout": 300,
+  "lambda_memory_size": 2560
```
[the remark this answers is not in the document]
Comment (xbrianh) [reply daemons/dss-events-scribe/.chalice/config.json]: We can consider reducing allocations for this and other daemons.

**Pull request merged**